### ISSUE TITLE
Install script assumes Homebrew installed to /usr/local

### ISSUE THREAD

**Issue by nateberkopec** [oracle/truffleruby#1583]
Mine is installed in my home directory, so this check:

```ruby 
if macOS && !ENV['OPENSSL_PREFIX']
  if Dir.exist?('/usr/local/opt/openssl') # Homebrew
    ENV['OPENSSL_PREFIX'] = '/usr/local/opt/openssl'
```

...fails.

Separately, even after correctly setting OPENSSL_PREFIX, I still get: 

```
checking for openssl/ssl.h... no
```

**Comment by chrisseaton** [comment]
Does `$ brew config` give the correct directory for `HOMEBREW_PREFIX`? If so we could use that.

**Comment by nateberkopec** [comment]
It does, and so does `brew --prefix openssl`

**Comment by eregon** [comment]
@nateberkopec Just to be sure, what did you set `OPENSSL_PREFIX` to?

Did you use a Ruby manager/installer to install TruffleRuby, which one?

Just as a note, Homebrew recommends against using anything else than `/usr/local` (https://docs.brew.sh/Installation), as it means other software cannot find Homebrew packages easily like here. We try to fix bugs for other systems/setups, but in the end we can only test a few system configurations.

Probably this line (from the original `extconf.rb` in MRI) fails:
```ruby
result = pkg_config("openssl") && have_header("openssl/ssl.h")
```

**Comment by nateberkopec** [comment]
I realize not installing to /usr/local is nonstandard, but this is actually the first trouble I've ever encountered with it.

> Probably this line (from the original `extconf.rb` in MRI) fails:

FWIW MRI installs just fine.

I set `OPENSSL_PREFIX=/Users/nateberkopec/.homebrew/opt/openssl`

**Comment by nateberkopec** [comment]
And I'm using `ruby-install`, so the whole command is `OPENSSL_PREFIX=/Users/nateberkopec/.homebrew/opt/openssl ruby-install truffleruby`

**Comment by eregon** [comment]
> FWIW MRI installs just fine.

Interesting, and that seems to work due to special handling of `brew` in `ruby-install`, which I didn't notice: https://github.com/postmodern/ruby-install/blob/ea2b8bbff87ea4340d965e72ac3acfb4ab1d1766/share/ruby-install/ruby/functions.sh#L24

We can definitely call `brew --prefix openssl` and set `OPENSSL_PREFIX` from it, the only remaining question is how to let `have_header` search that.

**Comment by eregon** [comment]
I tried on Linux when uninstalling my system OpenSSL and  building OpenSSL in a custom prefix, and everything seems to work as expected.

For the record, on Linux, building openssl from source is:
```
$ ./config --prefix=$HOME/prefix/openssl '-Wl,-rpath,$(LIBRPATH)'
$ make -j4
$ make install
```
(Without the -rpath, ` ldd .../lib/libssl.so` wouldn't find the correct `libcrypto.so` and use the system one)

@nateberkopec Could you run this, and share
`~/.rubies/truffleruby-1.0.0-rc12/src/main/c/openssl/mkmf.log` in a gist?
```
rm -r ~/.rubies/truffleruby-1.0.0-rc12
OPENSSL_PREFIX="$(brew --prefix openssl)" ruby-install truffleruby
```

**Comment by nateberkopec** [comment]
Nice. Setting `OPENSSL_PREFIX="$(brew --prefix openssl)"` fixed it, so I think that can close up this issue.

The *other* issue I was having is [explained here:](https://forums.developer.apple.com/thread/104296#317543)

> The Command Line Tools package installs the macOS system headers inside the macOS SDK. Software that compiles with the installed tools will search for macOS headers at the standard include path: /Applications/Xcodebeta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.1 4.sdk/usr/include For legacy software that looks for the macOS headers in the base system under /usr/include, please install the package file located at: /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14. pkg

Running that package fixed the issue and truffleruby installed fine.

**Comment by chrisseaton** [comment]
We mention that second bit for the LLVM install https://github.com/oracle/truffleruby/blob/master/doc/user/installing-llvm.md#macos. Hopefully it will become common knowledge soon, as you need it for everything.

**Comment by chrisseaton** [comment]
But we should still detect your non-standard prefix automatically, right? You shouldn't have to use `OPENSSL_PREFIX` should you?

**Comment by nateberkopec** [comment]
> We mention that second bit for the LLVM install

Yikes! I skipped right over that and went for the `brew install llvm@4` and thought "yes, that's done, all good now."

> But we should still detect your non-standard prefix automatically, right? You shouldn't have to use `OPENSSL_PREFIX` should you?

Correct. If I don't set OPENSSL_PREFIX myself, it blows up on that step. I think you should just rewrite the script to use brew --prefix (and deal with `brew` not existing for other package manager users).

**Comment by eregon** [comment]
@nateberkopec I fixed this in 01fa8d26473125d8e966418898c61a2818cc2d32, it will be in the next release. Thank you for the report!
